packages/core/src/services/zoom.ts

```diff
@@ -67,11 +67,21 @@ export class Zoom extends Service {
 		const mainXAxisPosition = cartesianScales.getMainXAxisPosition();
 		const domainIdentifier = cartesianScales.getDomainIdentifier();
 
-		// default to full range with extended domain
-		return cartesianScales.extendsDomain(
+		const customDomain = Tools.getProperty(
+			this.model.getOptions(),
+			'axes',
 			mainXAxisPosition,
-			extent(allZoomBarData, (d: any) => d[domainIdentifier])
+			'domain'
 		);
+
+		// Return custom domain if defined
+		// default to full range with extended domain
+		return customDomain
+			? customDomain
+			: cartesianScales.extendsDomain(
+					mainXAxisPosition,
+					extent(allZoomBarData, (d: any) => d[domainIdentifier])
+			  );
```

Comment:
we should probably do some checks here and make sure the custom domain is correctly formatted... e.g. the provided custom domain isn't a string, or an int, but an array of size 2...?

Reply:
this should probably just be a function somewhere that we can re-use since we pull in the custom domains in multiple places

Reply:
> this should probably just be a function somewhere that we can re-use since we pull in the custom domains in multiple places

From my findings, custom domain is only retrieved twice, one in `zoom.ts` & second in `axis.ts`
```ts
Tools.getProperty(
       options,
       'axes',
       axisPosition,
       'domain'
);
```